commit adb69f7fd42245c8193b930306880057d06f1df0
Author: Andrey Danin <danindrey@mail.ru>
Date:   Mon Jan 21 11:22:33 2013 +0400

    cbootimage: new package

diff --git a/package/Config.in b/package/Config.in
index 7b83b8b..cd1d5fe 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -198,6 +198,7 @@ endmenu
 menu "Hardware handling"
 source "package/abootimg/Config.in"
 source "package/acpid/Config.in"
+source "package/cbootimage/Config.in"
 source "package/cdrkit/Config.in"
 source "package/dbus/Config.in"
 source "package/dbus-glib/Config.in"
diff --git a/package/cbootimage/Config.in b/package/cbootimage/Config.in
new file mode 100644
index 0000000..e964fd8
--- /dev/null
+++ b/package/cbootimage/Config.in
@@ -0,0 +1,10 @@
+config BR2_PACKAGE_CBOOTIMAGE
+	bool "cbootimage"
+	help
+	  cbootimage can compile a text representations of a BCT into a
+	  binary, or combine a binary and a bootloader into a flashable
+	  image.
+	  
+	  https://gitorious.org/cbootimage/cbootimage
+	  http://nv-tegra.nvidia.com/gitweb/?p=tools/cbootimage.git;a=summary	  
+
diff --git a/package/cbootimage/cbootimage-compilation.patch b/package/cbootimage/cbootimage-compilation.patch
new file mode 100644
index 0000000..e673141
--- /dev/null
+++ b/package/cbootimage/cbootimage-compilation.patch
@@ -0,0 +1,67 @@
+diff --git a/data_layout.c b/data_layout.c
+index a0b2e11..ae154a8 100644
+--- a/data_layout.c
++++ b/data_layout.c
+@@ -92,13 +92,12 @@ iceil_log2(u_int32_t a, u_int32_t b)
+ static u_int32_t
+ ceil_log2(u_int32_t a)
+ {
+-	u_int32_t result;
+-
+-	result = log2(a);
+-	if ((1UL << result) < a)
+-		result++;
+-
+-	return result;
++	int  e;
++	//ASSERT (a >= 1);
++	for (e = 0; ; e++)
++		if ((1 << e) >= a)
++			break;
++	return e;
+ }
+ 
+ static void init_bad_block_table(build_image_context *context)
+diff --git a/set.c b/set.c
+index 70c9522..a248573 100644
+--- a/set.c
++++ b/set.c
+@@ -181,6 +181,20 @@ context_set_array(build_image_context	*context,
+ 	return 0;
+ }
+ 
++static inline int32_t ilog2(u_int32_t x)
++{
++	int  e;
++	//ASSERT (a >= 1);
++	for (e = 0; ; e++)
++		if ((1 << e) >= x)
++			break;
++
++	if ((1 << e) > x)
++		e -= 1;
++
++	return e;
++}
++
+ /*
+  * context_set_value(): General handler for setting values in config files.
+  */
+@@ -197,7 +211,7 @@ int context_set_value(build_image_context *context,
+ 
+ 	case token_block_size:
+ 		context->block_size = value;
+-		context->block_size_log2 = log2(value);
++		context->block_size_log2 = ilog2(value);
+ 
+ 		if (context->memory != NULL) {
+ 			printf("Error: Too late to change block size.\n");
+@@ -225,7 +239,7 @@ int context_set_value(build_image_context *context,
+ 
+ 	case token_page_size:
+ 		context->page_size = value;
+-		context->page_size_log2 = log2(value);
++		context->page_size_log2 = ilog2(value);
+ 		context->pages_per_blk= 1 << (context->block_size_log2- 
+ 			context->page_size_log2);
+ 
diff --git a/package/cbootimage/cbootimage.mk b/package/cbootimage/cbootimage.mk
new file mode 100644
index 0000000..6d92d80
--- /dev/null
+++ b/package/cbootimage/cbootimage.mk
@@ -0,0 +1,21 @@
+#############################################################
+#
+# cbootimage
+#
+#############################################################
+
+CBOOTIMAGE_VERSION = 20379c18a25191d9197bb96fd719c99336100055
+CBOOTIMAGE_SITE = git://gitorious.org/cbootimage/cbootimage.git
+CBOOTIMAGE_DEPENDENCIES = util-linux
+
+define CBOOTIMAGE_BUILD_CMDS
+	$(MAKE) CC="$(TARGET_CC)" lD="$(TARGET_LD)" CFLAGS=-Os -C $(@D) all
+endef
+
+define CBOOTIMAGE_INSTALL_TARGET_CMDS
+	$(INSTALL) -m 0755 $(@D)/cbootimage $(TARGET_DIR)/usr/bin
+	$(INSTALL) -m 0755 $(@D)/bct_dump $(TARGET_DIR)/usr/bin
+endef
+
+$(eval $(generic-package))
+
