commit e0abecb7432a20efebddb5bc5cdf1f9da2104e81
Author: Andrey Danin <danindrey@mail.ru>
Date:   Wed Jan 16 12:12:24 2013 +0400

    abootimg: new package (adopted Gilles' Grandou patch)

diff --git a/package/Config.in b/package/Config.in
index 15d7ad6..7b83b8b 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -196,6 +196,7 @@ source "package/xfsprogs/Config.in"
 endmenu
 
 menu "Hardware handling"
+source "package/abootimg/Config.in"
 source "package/acpid/Config.in"
 source "package/cdrkit/Config.in"
 source "package/dbus/Config.in"
diff --git a/package/abootimg/Config.in b/package/abootimg/Config.in
new file mode 100644
index 0000000..5b20ea6
--- /dev/null
+++ b/package/abootimg/Config.in
@@ -0,0 +1,19 @@
+config BR2_PACKAGE_ABOOTIMG
+	bool "abootimg"
+	select BR2_PACKAGE_UTIL_LINUX
+	select BR2_PACKAGE_UTIL_LINUX_LIBBLKID
+	select BR2_PACKAGE_UTIL_LINUX_LIBUUID
+	help
+	  Tool to read/write/update android boot images
+	  
+	  Android devices use a special partition format to boot any
+	  operating system on the devices. These boot-images contain
+	  a kernel image, a ramdisk, optionally a 2nd stage boot loader
+	  and the commandline passed to the kernel when booting.
+	  The original mkbootimg from Android can only create these images
+	  where abootimg can also extract and modify them.
+	  Handling android boot images is necessary when bringing other
+	  operating systems to android devices.
+	  
+	  http://gitorious.org/ac100/abootimg
+
diff --git a/package/abootimg/abootimg.mk b/package/abootimg/abootimg.mk
new file mode 100644
index 0000000..eabd648
--- /dev/null
+++ b/package/abootimg/abootimg.mk
@@ -0,0 +1,23 @@
+#############################################################
+#
+# abootimg
+#
+#############################################################
+
+ABOOTIMG_VERSION = v0.6
+ABOOTIMG_SITE = git://gitorious.org/ac100/abootimg.git
+ABOOTIMG_DEPENDENCIES = util-linux
+
+define ABOOTIMG_BUILD_CMDS
+	echo "#define VERSION_STR \"$(ABOOTIMG_VERSION)\"" > $(@D)/version.h
+	$(MAKE) CC="$(TARGET_CC)" lD="$(TARGET_LD)" CFLAGS=-Os -C $(@D) all
+endef
+
+define ABOOTIMG_INSTALL_TARGET_CMDS
+	$(INSTALL) -m 0755 $(@D)/abootimg $(TARGET_DIR)/usr/bin
+	$(INSTALL) -m 0755 $(@D)/pack-initrd $(TARGET_DIR)/usr/bin/abootimg-pack-initrd
+	$(INSTALL) -m 0755 $(@D)/unpack-initrd $(TARGET_DIR)/usr/bin/abootimg-unpack-initrd
+endef
+
+$(eval $(generic-package))
+
